build.sbt

```diff
@@ -137,32 +147,24 @@ lazy val assemblySettings = Seq(
     case s if s.endsWith("snappyjava_snappy.dll") => MergeStrategy.last
     case s if s.endsWith(".dtd")                  => MergeStrategy.rename
     case s if s.endsWith(".xsd")                  => MergeStrategy.rename
+    case PathList("META-INF", "NOTICE")           =>
+      // avro-tools META-INF/NOTICE must not be renamed
+      CustomMergeStrategy.rename {
+        case l: Assembly.Library if l.moduleCoord.name == "avro-tools" =>
+          l.target
+        case l: Assembly.Library =>
+          l.target + "_" + l.moduleCoord.name + "-" + l.moduleCoord.version
+        case p: Assembly.Project =>
+          p.target + "_" + p.name
+      }
     case PathList("META-INF", "services", "org.apache.hadoop.fs.FileSystem") =>
       MergeStrategy.filterDistinctLines
-    case PathList("META-INF", "LICENSE")     => MergeStrategy.discard
-    case PathList("META-INF", "MANIFEST.MF") => MergeStrategy.discard
-    case PathList("META-INF", "INDEX.LIST")  => MergeStrategy.discard
+    case PathList("META-INF", "LICENSE")               => MergeStrategy.discard
+    case PathList("META-INF", "MANIFEST.MF")           => MergeStrategy.discard
+    case PathList("META-INF", "INDEX.LIST")            => MergeStrategy.discard
     case PathList("META-INF", s) if s.endsWith(".DSA") => MergeStrategy.discard
     case PathList("META-INF", s) if s.endsWith(".RSA") => MergeStrategy.discard
     case PathList("META-INF", s) if s.endsWith(".SF")  => MergeStrategy.discard
-    case PathList("META-INF", "NOTICE")      => MergeStrategy.rename
-    case _                                   => MergeStrategy.last
+    case _                                             => MergeStrategy.last
   })
 )
-
-import sbtassembly.AssemblyUtils
-import sbtassembly.MergeStrategy
-
-def jarFilter(_name: String)(f: File => Boolean): MergeStrategy = new MergeStrategy {
```

Comment:
This packaging rule is obscure.
it is greatly improved with `sbt-asembly-2.0.0`